source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,215 @@
+
+.. raw:: html
+
+   <h4>Battery Sag Calculator</h4>
+   <div style="width:100%; overflow:hidden;">
+      <form onkeypress="return event.keyCode != 13" style="float:left;">
+         <p>Unloaded voltage (V)</p>
+         <input onchange="updateOutput()" id="uV" value="12.5" style="width:90%;" type="numeric" placeholder="12.5"/>
+      </form>
+      <form onkeypress="return event.keyCode != 13" style="float:left;">
+         <p>Total draw (A)</p>
+         <input onchange="updateOutput()" id="current" value="240" style="width:90%;" type="numeric" placeholder="240"/>
+      </form>
+      <form onkeypress="return event.keyCode != 13" style="float:left;">
+         <p>Battery resistance (mOhms)</p>
+         <input onchange="updateOutput()" id="resistance" value="20" style="width:90%;" type="numeric" placeholder="20"/>
+      </form>
+      <p style="float:left;margin-left:10px;margin-top:35px;font-weight:bold;color:#bdeb34;">=<span id="output">10.12V</span></p>
+   </div>
+   <br/>
+
+   <script>
+      updateOutput();
+
+      function updateOutput() {
+         var unloadedVoltage = document.getElementById("uV").value
+         var current = document.getElementById("current").value
+         var resistance = document.getElementById("resistance").value
+         var output = document.getElementById("output")
+
+         var calculatedOutput = parseFloat(unloadedVoltage) - (parseFloat(current) * (parseFloat(resistance) / 1000))
+
+         output.innerHTML = (Math.round(calculatedOutput*10**2)/10**2) + "V"
+      }
+   </script>
+
+Be aware that battery health (in the form of battery resistance above) changes how much increased current draw effects the output voltage of the battery. Health of the battery can be roughly determined via a `battery beak <https://store.ctr-electronics.com/battery-beak/>`__ or a via a battery discharge test with a `battery analyzer <https://www.andymark.com/products/computerized-battery-analyzer>`__.
+
+Using the above information, ensure your battery is healthy and that your supply current limits will prevent the battery sagging below 7V.
+
+Using Limits to Reduce Wheel Slip
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Stator current limits are excellent at preventing wheel slip (thus increasing traction). To determine wheel slip, perform the following instructions.
+
+1. Place the robot on carpet against a wall.
+2. Begin plotting velocity and stator current in :doc:`Tuner X </docs/tuner/plotting>`.
+3. Slowly increase dutycycle until the velocity becomes non-zero.
+
+Set your stator limit to a value below the observed stator current in Tuner. In the below plot, you can see that the wheels began slipping at around 130A.
+
+.. image:: images/slip-current.png
+   :alt: wheel slip at 130A
+
+An interesting observation here is that supply current in the above graph was only around ~37A, this information helps provide real world evidence that stator current limits are effective at reducing brownouts.
```

Comment:
I don't like this, that current is the *minimum* supply current while the motor is below ~4500 RPM.

Reply:
to be clear, because the change in velocity in the graph is very low?

Reply:
Yep, that's essentially supply current at stall. But for a stator current limit, supply current starts at its lowest and increases as motor velocity increases.

Reply:
I'm working on getting a better plot. Would a plot that shows a drastic change in rotation velocity work?

Reply:
The slip current plot is fine, it's just the statement after that's not quite correct. I would just remove it entirely and not plot supply current.